lib/src/generators/assets_generator.dart

```diff
@@ -3,66 +3,53 @@ import 'dart:io';
 
 import '../settings/asset_type.dart';
 import '../settings/flutter.dart';
 import '../settings/flutter_gen.dart';
-import '../utils/camel_case.dart';
+import '../utils/string.dart';
 import 'generator_helper.dart';
 import 'integrations/integration.dart';
 import 'integrations/svg_integration.dart';
 
-String generateAssets(File pubspecFile, DartFormatter formatter,
-    FlutterGen flutterGen, FlutterAssets assets) {
+String generateAssets(
+  File pubspecFile,
+  DartFormatter formatter,
+  FlutterGen flutterGen,
+  FlutterAssets assets,
+) {
   assert(assets != null && assets.hasAssets,
       throw 'The value of "flutter/assets:" is incorrect.');
 
   final importsBuffer = StringBuffer();
   final classesBuffer = StringBuffer();
 
-  final assetRelativePathList = _getAssetRelativePathList(pubspecFile, assets);
-  final assetTypeQueue = ListQueue<AssetType>.from(
-      _constructAssetTree(assetRelativePathList).children);
-  final assetsStaticStatements = <_Statement>[];
-
   final integrations = <Integration>[];
   if (flutterGen != null &&
       flutterGen.hasIntegrations &&
       flutterGen.integrations.hasFlutterSvg) {
     integrations.add(SvgIntegration());
   }
 
-  while (assetTypeQueue.isNotEmpty) {
-    final assetType = assetTypeQueue.removeFirst();
-    final assetAbsolutePath = join(pubspecFile.parent.path, assetType.path);
-
-    if (FileSystemEntity.isDirectorySync(assetAbsolutePath)) {
-      final statements = _createDirectoryClassGenStatements(
-          pubspecFile, assetType, integrations);
-
-      if (assetType.isDefaultAssetsDirectory) {
-        assetsStaticStatements.addAll(statements);
-      } else {
-        final className = '\$${assetType.path.camelCase().capitalize()}Gen';
-        classesBuffer
-            .writeln(_directoryClassGenDefinition(className, statements));
-        // Add this directory reference to Assets class
-        // if we are not under the default asset folder
-        if (dirname(assetType.path) == '.') {
-          assetsStaticStatements.add(_Statement(
-            type: className,
-            name: assetType.baseName.camelCase(),
-            value: '$className\(\)',
-            isConstConstructor: true,
-          ));
-        }
-      }
+  String assetsGenStyle;
+  if (flutterGen != null && flutterGen.hasAssets) {
+    assetsGenStyle = flutterGen.assets.style;
+  }
 
-      assetTypeQueue.addAll(assetType.children);
-    }
+  if (assetsGenStyle == 'dot-delimiter' || assetsGenStyle == null) {
+    classesBuffer.writeln(
+        _dotDelimiterStyleDefinition(pubspecFile, assets, integrations));
+  } else if (assetsGenStyle == 'snake-case') {
+    classesBuffer
+        .writeln(_snakeCaseStyleDefinition(pubspecFile, assets, integrations));
+  } else if (assetsGenStyle == 'camel-case') {
+    classesBuffer
+        .writeln(_camelCaseStyleDefinition(pubspecFile, assets, integrations));
+  } else {
+    throw 'The value of "flutter_gen/assets/style." is incorrect.';
```

Comment:
And.

It doesn't work on my mac when style is camel-case or snake-case.
```
FlutterGen Loading ... example/pubspec.yaml
Unhandled exception:
NoSuchMethodError: The method 'toStaticFieldString' was called on null.
Receiver: null
Tried calling: toStaticFieldString()
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
#1      _assetsClassDefinition.<anonymous closure> (package:flutter_gen/src/generators/assets_generator.dart:275:31)
#2      MappedListIterable.elementAt (dart:_internal/iterable.dart:417:31)
#3      ListIterable.join (dart:_internal/iterable.dart:149:25)
#4      _assetsClassDefinition (package:flutter_gen/src/generators/assets_generator.dart:277:8)
#5      _flatStyleDefinition (package:flutter_gen/src/generators/assets_generator.dart:269:10)
#6      _camelCaseStyleDefinition (package:flutter_gen/src/generators/assets_generator.dart:225:10)
#7      generateAssets (package:flutter_gen/src/generators/assets_generator.dart:48:18)
#8      FlutterGenerator.build (package:flutter_gen/flutter_generator.dart:67:27)
<asynchronous suspension>
#9      main (file:///Users/a12622/git/flutter_gen/bin/flutter_gen_command.dart:37:48)
#10     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:299:32)
#11     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)
```

Reply:
Fixed!